bench/bench_test.go

```diff
@@ -187,157 +174,98 @@ func BenchmarkParallelGetSet(b *B) {
 	// connections needed for the benchmarks.
 	poolSize := parallel * runtime.GOMAXPROCS(0)
 
-	do := func(b *B, fn func()) {
+	do := func(b *B, fn func() error) {
 		b.ResetTimer()
 		b.SetParallelism(parallel)
 		b.RunParallel(func(pb *PB) {
 			for pb.Next() {
-				fn()
+				if err := fn(); err != nil {
+					b.Fatal(err)
+				}
 			}
 		})
 	}
 
 	b.Run("radix", func(b *B) {
-		b.Run("no pipelining", func(b *B) {
-			pool, err := radix.NewPool("tcp", "127.0.0.1:6379", poolSize, radix.PoolPipelineWindow(0, 0))
-			if err != nil {
-				b.Fatal(err)
-			}
-			defer pool.Close()
-			if err := waitFull(pool, poolSize); err != nil {
-				b.Fatal(err)
-			}
+		mkRadixBench := func(opts ...radix.PoolOpt) func(b *B) {
+			return func(b *B) {
+				pool, err := radix.NewPool("tcp", "127.0.0.1:6379", poolSize, opts...)
+				if err != nil {
+					b.Fatal(err)
+				}
+				defer pool.Close()
 
-			do(b, func() {
-				err := pool.Do(radix.WithConn("foo", func(conn radix.Conn) error {
-					if err := conn.Do(radix.Cmd(nil, "SET", "foo", "bar")); err != nil {
+				// wait for the pool to fill up
+				for {
+					time.Sleep(50 * time.Millisecond)
+					if pool.NumAvailConns() >= poolSize {
+						break
+					}
+				}
+
+				do(b, func() error {
+					if err := pool.Do(radix.Cmd(nil, "SET", "foo", "bar")); err != nil {
```

Comment:
Maybe replace this with a call to `radixGetSet`?

Reply:
The problem is that `radixGetSet` takes a Conn, not a Client, cause I was worried about the overhead of converting the types to interfaces.

But now that I think about it, I could pre-convert them before the bench starts, and that shouldn't incur any overhead...